include/libembeddedhal/adc/adc_stub.hpp

```diff
@@ -0,0 +1,20 @@
+#include "adc.hpp"
+
+namespace embed::stub {
+/**
+ * @brief A stub implementation of embed::adc to be used as a placeholder for
+ * drivers that require an optional adc. Calling read() will return a
+ * percent with a value of 50%. This implementaion should only be used when a
+ * driver has stated in its documentation that the adc is optional and can be
+ * stubbed out.
+ *
+ */
+class adc : public embed::adc
+{
+private:
+  boost::leaf::result<percent> driver_read() noexcept
+  {
+    return embed::percent::from_ratio(1, 2);
```

Comment:
```suggestion
    return m_adc_value;
```

I think it would make sense if the ADC took a default value in its constructor then always returned that. Because half voltage could mean something very specific.